PORTING.md

```diff
@@ -0,0 +1,327 @@
+You can leave the `defines()` and `appendInterrupt()` macros as they are, but you need to implement the `startupCode()` macro with whatever platform specific hardware needs to be initialized. Don't do complicated stuff here, the system is not fully booted yet!
+
+The [simplest linkerscript available is `stm32/stm32_ram.ld.in`](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/driver/core/cortex/stm32/stm32_ram.ld.in?ts=4), which fits everything into one SRAM region.
+The linkerscipt diagram in the comment gives you an overview of what goes where, check that against the devices memory map. Note that while the diagram shows specific addresses, the section addresses and sizes are taken from the device file!
+
+You need to tell the [core driver which linkerscript to use for you device](https://github.com/roboterclubaachen/xpcc/blob/develop/src/xpcc/architecture/platform/driver/core/cortex/driver.xml?ts=4#L29-L40):
+```xml
+<template device-platform="sam" out="linkerscript.ld">stm32/stm32_ram.ld.in</template>
+```
+
+### Compile an empty example
+
+At this point you should test your changes to the build system and the port of the core driver.
+Create an empty example in `xpcc/examples/sam4/empty`, containing:
+
+- the default `SConstruct` file:
+```python
+# path to the xpcc root directory
+xpccpath = '../../..'
+# execute the common SConstruct file
+execfile(xpccpath + '/scons/SConstruct')
+```
+
+- an empty `main.cpp` file:
+```cpp
+// your copyright with bsd license
+#include <xpcc/architecture/platform.hpp>
+
+int main()
+{
+    while(1) { /* ... tumbleweed ... */ }
+}
+```
+
+- the `project.cfg` file:
+```ini
+[build]
+device = sam4lc4b
+buildpath = ${xpccpath}/build/sam4/${name}
+```
+
+To compile this example, `cd` into the folder and execute `scons`.
+If everything goes well, there will be warnings about the missing `clock` and `gpio` drivers, but at the end there should be a printout of the resource usage, similar to this (but with realistic numbers):
+```sh
+cd examples/sam4/empty
+scons
+[...]
+Memory Usage
+------------
+Device: sam4lc4b
+
+Program:    1000 bytes (1.0% used)
+(.data + .reset + .rodata + .text)
+
+Data:       1000 bytes (2.0% used) = 1 bytes static (1.0%) + 1000 bytes stack (1.0%)
+(.bss + .data + .noinit + .stack)
+
+Heap:      9999 bytes (97.0% available)
+(.heap1)
+```
+
+You can explore the generated code for the core driver in `xpcc/build/sam4/empty/libxpcc/generated_platform/driver/core/cortex/`.
+
+We'll defer flashing onto the device, until we can actually blink an LED.
+Without a debugger you won't be able to tell what is going.
+
+If something goes wrong in the building step, check the previous steps, maybe you missed something, or we forgot to add something to this guide.
+If you're totally stuck write us an email (try to work the problem first though ;-).
+
+### Porting the clock driver
```

Comment:
A more natural order would imho be to first port the GPIO driver.
It is much more straight forward compared to the clock driver and it actually provides you with a visible result.